cogs/tags/tags.py

```diff
@@ -119,9 +120,29 @@ def tag_decoder(obj):
         return TagAlias(**obj)
     return obj
 
+def roles_or_mod_or_permissions(all_server_roles: dict = {}, **perms):
+    async def predicate(ctx):
+        server = ctx.guild
+        mod_roles = [r.name.lower() for r in await ctx.bot.get_mod_roles(server)]
+        admin_roles = [r.name.lower() for r in await ctx.bot.get_admin_roles(server)]
+        roles = mod_roles + admin_roles + list(all_server_roles[server])
+        return await role_or_permissions(ctx, lambda r: r.name.lower() in roles, **perms)
+    return commands.check(predicate)
```

Comment:
Do you want to replace the one in `rolechecks.py`?
1. Index by server ID. Also might want to rename server to guild, but can do that afterwards.
2. Need to handle case where no roles are configured.
3. Check names, no IDs.
```suggestion
def roles_or_mod_or_permissions(allowed_roles: dict={}, **perms):
    async def predicate(ctx):
        server = ctx.guild
        server_roles = []
        if server:
            server_roles = allowed_roles.get(server.id, [])
        mod_roles = [r.name.lower() for r in await ctx.bot.get_mod_roles(server)]
        admin_roles = [r.name.lower() for r in await ctx.bot.get_admin_roles(server)]
        roles = mod_roles + admin_roles + server_roles
        return await role_or_permissions(ctx, lambda r: r.name.lower() in roles, **perms)
    return commands.check(predicate)
```

Reply:
I don't understand 2 and 3. For 2 do you mean if allowed_roles is passed without an entry for this server and that isn't a defaultdict? For 3, aren't the roles retrieved from the dict already the role names? Or did you mean a different ID?

Reply:
I'd rather not replace role_or_mod_or_permissions because this function requires a dict of guilds to be passed instead of a single role name.

Reply:
For 2, the following line will fail because of a `KeyError`:
```
roles = mod_roles + admin_roles + list(all_server_roles[server])
```
For 3, `mod_roles` and `admin_roles` are role names, while `all_server_roles` are role IDs.

Reply:
Re: 2, in that case, make the default for `all_server_roles` a defaultdict instead.